quint/src/runtime/compile.ts

```diff
@@ -175,8 +177,17 @@ export function compileDef(
 ): CompilationContext {
   // Define a new module list with the new definition in the last module,
   // ensuring the original object is not modified
-  const modules: QuintModule[] = [...state.modules] // Those are flat, but introducing `def` might make them non-flat
-  const lastModule: FlatModule = state.modules[state.modules.length - 1] // This is not modules.pop() to ensure flatness
+  const originalModules = [...state.originalModules]
```

Comment:
It looks like you could use `splice` here, instead of copying the array and modifying it

Reply:
We cannot change the original array, so I'm not seeing how `splice` would help here.